### PR TITLE
eth/catalyst: set random value in dev mode

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR sets the random field in catalyst. This mimics the behavior of the chain better than having it 0 all the time

### CODE REVIEW

**Review by jwasinger — APPROVED** [review]
LGTM

**Review by holiman — APPROVED** [review]
The change is fine with me, but I have a bit of a feeling that maybe we will regret adding non-determinism to the dev chain like this...? Nothing tangible, but for example it might be more difficult (in general) to repro reported bugs. 
An alternative would be to init an `rng` and use that rng exclusively for the randoms in the FCU. 
Since I have nothing really tangible, I still click Approve though.

File: eth/catalyst/simulated_beacon.go [hunk 1]
```diff
@@ -143,16 +144,19 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
...
 
-	// Reset to CurrentBlock in case of the chain was rewound
+  // Reset to CurrentBlock in case of the chain was rewound
 	if header := c.eth.BlockChain().CurrentBlock(); c.curForkchoiceState.HeadBlockHash != header.Hash() {
 		finalizedHash := c.finalizedBlockHash(header.Number.Uint64())
 		c.setCurrentState(header.Hash(), *finalizedHash)
 	}
 
+  var random [32]byte
+	rand.Read(random[:])
```
Comment (holiman) [inline eth/catalyst/simulated_beacon.go:154]:
```suggestion
	var random [32]byte
	rand.Read(random[:])
```

File: eth/catalyst/simulated_beacon.go [hunk 2]
```diff
@@ -143,16 +144,19 @@ func (c *SimulatedBeacon) sealBlock(withdrawals []*types.Withdrawal) error {
 	feeRecipient := c.feeRecipient
 	c.feeRecipientLock.Unlock()
 
-	// Reset to CurrentBlock in case of the chain was rewound
+  // Reset to CurrentBlock in case of the chain was rewound
```
Comment (holiman) [inline eth/catalyst/simulated_beacon.go:147]:
```suggestion
	// Reset to CurrentBlock in case of the chain was rewound
```

**Review by karalabe — APPROVED** [review]
SGTM

**Pull request merged**